# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1285865603 14400
#      Thu Sep 30 12:53:23 2010 -0400
# Node ID d7f0d115c10c509a9d17f7210c6932362d9f5430
# Parent  e678346a47d94855f59489bb4dc75232d5e0c817
pt-idx.cc (tree_expression::lvalue): use octave_map here, not octave_scalar_map

diff --git a/src/pt-idx.cc b/src/pt-idx.cc
--- a/src/pt-idx.cc
+++ b/src/pt-idx.cc
@@ -525,25 +525,27 @@ tree_index_expression::lvalue (void)
                 bool autoconv = (tmp.is_zero_by_zero () 
                                  && (tmp.is_matrix_type () || tmp.is_string ()
                                      || tmp.is_cell ()));
 
                 if (i > 0 && type [i-1] == '(')
                   {
                     octave_value_list pidx = idx.back ();
 
+                    // Use octave_map, not octave_scalar_map so that the
+                    // dimensions are 0x0, not 1x1.
                     if (tmp.is_undefined ())
                       {
                         if (pidx.has_magic_colon ())
                           gripe_invalid_inquiry_subscript ();
                         else
-                          tmp = octave_scalar_map ();
+                          tmp = octave_map ();
                       }
                     else if (autoconv)
-                      tmp = octave_scalar_map ();
+                      tmp = octave_map ();
 
                     retval.numel (tmp.numel (pidx));
 
                     tmpi = i-1;
                     tmpidx.push_back (tidx);
                   }
                 else
                   {
